app/frontend/wallet/helpers/generateDeriveXpubFn.js

```diff
@@ -0,0 +1,38 @@
+const indexIsHardened = require('./indexIsHardened')
+const {derivePublic} = require('cardano-crypto.js')
+
+function generateDeriveXpubFn(deriveXpubHardenedFn, derivationScheme) {
+  const derivedXpubs = {}
+
+  async function deriveXpub(absDerivationPath) {
+    const memoKey = JSON.stringify(absDerivationPath)
+
+    if (!derivedXpubs[memoKey]) {
+      const deriveHardened =
+        absDerivationPath.length === 0 ||
+        indexIsHardened(absDerivationPath[absDerivationPath.length - 1])
```

Comment:
Do we include lodash? If yes, use `_.last()`, if no, add `last` funciton to utilitities

Reply:
no, we don't,I used slice instead, although this can be considered hacky